src/App.jsx

```diff
@@ -61,14 +86,41 @@ class App extends Component {
       );
     }
 
+    if (this.state.isLoading) {
+      return (
+        <Provider>
+          <div className="container">
+            <FontAwesome spin name="cog" />
+          </div>
+        </Provider>
+      );
+    }
+
+    if (this.state.editSchema) {
+      return (
+        <Provider>
+          <Fixed m={2} right top>
+            <Button bg="red" onClick={() => this.logout()}>
+              Logout
+            </Button>
+          </Fixed>
+          <WrappedSchemaEditor store={store} done={() => this.setState({ editSchema: false })} />
+        </Provider>
+      );
+    }
+
     return (
       <Provider>
         <Fixed m={2} right top>
           <Button bg="red" onClick={e => this.logout()}>
             Logout
           </Button>
         </Fixed>
-        <DocumentViewer config={this.props.config} content={this.state.content} />
+        {schemaIsEditable &&
+          <div className="container">
+            <Button onClick={() => this.setState({ editSchema: true })}>Edit schema</Button>
+          </div>}
+        <WrappedDocumentViewer store={store} />
```

Comment:
I think this is a required prop for `react-redux`.
`Invariant Violation: Could not find "store" in either the context or props of "Connect(DocumentViewer)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(DocumentViewer)".`
Maybe it just looks weird because my store is called `store`